core/src/ledger_cleanup_service.rs

```diff
@@ -212,6 +212,8 @@ impl LedgerCleanupService {
                         lowest_cleanup_slot,
                         PurgeType::PrimaryIndex,
                     );
+                    // update only after purge operation
+                    blockstore.set_last_purged_slot(lowest_cleanup_slot);
```

Comment:
investigate when test `test_backup_and_clear_blockstore` fails if i move this into `blockstore_purge.rs`.